File: ClientApp/src/app/core/services/release-system.ts
import { Service, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { ReleaseSystem } from '../models';

@Service()
export class ReleaseSystemService {
  private readonly http = inject(HttpClient);

  listByRelease(releaseId: number) {
    return this.http.get<ReleaseSystem[]>(`/api/release-systems?releaseId=${releaseId}`);
  }

  create(releaseId: number, systemName: string, notes?: string) {
    return this.http.post<ReleaseSystem>('/api/release-systems', { releaseId, systemName, notes });
  }

  delete(id: number) {
    return this.http.delete<void>(`/api/release-systems/${id}`);
  }
}

File: Models/ReleaseSystem.cs
namespace rmp.Models;

public class ReleaseSystem
{
    public int Id { get; set; }
    public int ReleaseId { get; set; }
    public Release? Release { get; set; }
    public string SystemName { get; set; } = "";
    public string? Notes { get; set; }
}
